ta4j-core/src/main/java/org/ta4j/core/analysis/criteria/NumberOfLosingTradesCriterion.java

```diff
@@ -0,0 +1,44 @@
+package org.ta4j.core.analysis.criteria;
+
+import org.ta4j.core.PriceType;
+import org.ta4j.core.TimeSeries;
+import org.ta4j.core.Trade;
+import org.ta4j.core.TradingRecord;
+import org.ta4j.core.num.Num;
+import org.ta4j.core.num.PrecisionNum;
+
+/**
+ * Number of losing trades criterion.
+ */
+public class NumberOfLosingTradesCriterion extends AbstractBacktestingCriterion {
+
+    public NumberOfLosingTradesCriterion(PriceType priceType) {
+        super(priceType);
+    }
+
+    @Override
+    public Num calculate(TimeSeries series, TradingRecord tradingRecord) {
+         long numberOfLosingTrades = tradingRecord.getTrades().stream()
+                .filter(trade -> trade.isClosed())
+                .filter(trade -> isLosingTrade(series, trade)).count();
+         return PrecisionNum.valueOf(numberOfLosingTrades);
+    }
+
+    private boolean isLosingTrade(TimeSeries series, Trade trade) {
+        Num exitPrice = getPrice(series, trade.getExit());
+        Num entryPrice = getPrice(series, trade.getEntry());
+
+        Num profit = exitPrice.minus(entryPrice).multipliedBy(trade.getExit().getAmount());
+        return profit.isLessThan(PrecisionNum.valueOf(0));
+    }
+
+    @Override
+    public Num calculate(TimeSeries series, Trade trade) {
+        return series.numOf(1);
```

Comment:
I do not quite understand why it is better to cast an exception? The input parameter series and trade have value

Reply:
I think it should return 1 if the trade is a loosing trade and zero if the trade is a winning trade. Otherwise if you think this function makes no sense for just one trade it must be communicated to the user. At the moment this public function returns always 1. Is this the expected behaviour?

Reply:
Offcourse you are right. I have now fixed this isse :+1: